### PR TITLE
Check in gNMI 0.2.0 Specification

### PR DESCRIPTION
**Pull request by robshakir**
Add the initial public version of the gNMI specification, this includes the MarkDown and PDF specifications, alongside the proto definition file, and generated Python and Go code.

### CODE REVIEW

File: rpc/gnmi/gnmi-specification.md [hunk 1]
```diff
@@ -0,0 +1,955 @@
...
+
+### 3.4.1 The SetRequest Message 
+
+A `SetRequest` message consists of the following fields:
+
+
+
+*   `prefix` - specified as per [Section 2.4.1](#241-path-prefixes). The prefix specified is applied to all paths defined within other fields of the message.
+*   `delete` - A set of paths, specified as per [Section 2.2.2](#222-paths), which are to be removed from the data tree. A specification of the behavior of a delete is defined in [Section 3.4.5](#34-6-deleting-configuration).
```
Comment (robshakir) [inline rpc/gnmi/gnmi-specification.md]: Thanks, fixed. I ran a link checker over the doc, and couldn't find anything else like this.

File: rpc/gnmi/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# gNMI - gRPC Network Management Interface
+
+This directory contains a reference
+[protobuf3](https://github.com/google/protobuf/) service definition for the
+gRPC Network Management Interface (gNMI). This service provides an interface
```
Comment (robshakir) [inline rpc/gnmi/README.md]: Done.

File: rpc/gnmi/gnmi.proto [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+// Encoding defines the value encoding formats that are supported by the gNMI
+// protocol. These encodings are used by both the client (when sending Set
+// messages to modify the state of the target) and the target when serializing
+// data to be returned to the client (in both Subscribe and Get RPCs).
+// Reference: gNMI Specification Section 2.3
+enum Encoding {
+  JSON = 0;           // JSON encoded text.
+  BYTES = 1;          // Arbitrarily encoded bytes.
+  PROTO = 2;          // Encoded according to out-of-band agreed Protobuf.
+  JSON_IETF = 3;      // JSON encoded text as per RFC7951.
```
Comment (robshakir) [inline rpc/gnmi/gnmi.proto]: Either way we'll be left with an enum value that we can't use in the future, so I'm not sure I see a strong argument to change them around. Do you?

Comment (aashaikh) [inline rpc/gnmi/gnmi.proto]: yeah, i was just thinking that we'd have a gap for a while as I don't see us adding more encodings any time soon (maybe xml?)

Comment (robshakir) [inline rpc/gnmi/gnmi.proto]: I'll swap it around :-)

**Pull request merged**